### PR TITLE
Add a criminal records PDA cartridge

### PR DESCRIPTION
**Pull request by Brandon-Huu**
## About the PR

Todo:
- [ ] Adds a PDA cartridge that lists all of a stations criminal records ( excluding history )
  - [ ] **Improve the UI**
    - [x] Improve the UI
    - [ ] Add filters, ( wanted, detained )
    - [ ] Sort records in alpha-numeric order
    - [ ] Reuse controls from criminal records console
  - [ ] **Refactor the code** 2
    - [x] **Refactor the code**
    - [ ] use ``Entity<T>``
    - [ ] use caching?
  - [x] **Change the cartridge Icon**
  - [x] Add Localization
  
```
Separate PR:
Put a cartridge in warden's locker, so they can install the program onto PDAs of people who want it.
```
## Why / Balance

People fucking suck at responding to radio when asked "Who's wanted?"
Having to walk all the way back to sec to check records is extremely exhausting especially when the only reason you have to do it is because people refuse to communicate.

## Technical details

What the fuck does this code even do

## Media

Screenshot from 2024-03-10 14-31-55

Screenshot from 2024-03-10 14-29-49

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by Brandon-Huu** [comment]
I suck at making UI

**Comment by notquitehadouken** [comment]
> I suck at making UI

you are better at making ui than me

**Comment by Brandon-Huu** [comment]
I got the core functionality completed. So I think it's good enough to open for review.
Merge in under 10 minutes or we are no longer friends.

**Comment by Brandon-Huu** [comment]
Wait i forgot to push the new commits lmao

File: Content.Client/CartridgeLoader/Cartridges/CriminalRecordsCartridgeUiFragment.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            TextureScale = new Vector2(2f, 2f),
+            VerticalAlignment = VAlignment.Center,
+            Texture = _spriteSystem.Frame0(proto.Icon),
+            Margin = new Thickness(5, 0, 5, 0),
+        };
+
+        jobContainer.AddChild(jobIcon);
+        jobContainer.AddChild(jobLabel);
+    }
+
```
Comment (Brandon-Huu) [inline Content.Client/CartridgeLoader/Cartridges/CriminalRecordsCartridgeUiFragment.xaml.cs:147]: Is there a way to pass the Wanted & Detained control as a function parameter?

Comment (Plykiya) [reply Content.Client/CartridgeLoader/Cartridges/CriminalRecordsCartridgeUiFragment.xaml.cs:147]: Use the List typing that you made them out of?

Comment (Brandon-Huu) [reply Content.Client/CartridgeLoader/Cartridges/CriminalRecordsCartridgeUiFragment.xaml.cs]: > Use the List typing that you made them out of?

Thank you!

**Comment by Brandon-Huu** [comment]
I think the status label can also have an icon like the job label when this is merged #25192

**Comment by Dutch-VanDerLinde** [comment]
maybe security PDAs should have this by default instead of it being a cartridge?
cartridge should still exist, but only for promoted security members

File: Content.Server/CartridgeLoader/Cartridges/CrimnialRecordsCartridgeSystem.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<CriminalRecordsCartridgeComponent, CartridgeUiReadyEvent>(OnUiReady);
+    }
+
+    private void OnUiReady(
+      EntityUid uid,
+      CriminalRecordsCartridgeComponent criminalRecordsCartridge,
```
Comment (Brandon-Huu) [inline Content.Server/CartridgeLoader/Cartridges/CrimnialRecordsCartridgeSystem.cs:25]: In my last PR, @TheShuEd mentioned that I should use the ``Entity<T>`` syntax here. 
This is my first time working with systems like so I'm not entirely sure about why I would do that, or what I would have to change in the code. 

Any suggestions?

**Comment by beck-thompson** [comment]
Man I cannot wait for this to get merged. Its so difficult to tell who is wanted at the moment!

**Comment by juliangiebel** [comment]
You should probably share at least some of the ui controls between the criminal records console and the cartridge instead of writing it completely from scratch

File: Content.Server/CartridgeLoader/Cartridges/CrimnialRecordsCartridgeSystem.cs (hunk 2 above)
Comment (Tayrtahn) [reply Content.Server/CartridgeLoader/Cartridges/CrimnialRecordsCartridgeSystem.cs:25]: It's a pretty easy change.

```C#
private void OnUiReady(Entity<CriminalRecordsCartridgeComponent> entity, ref CartridgeUiReadyEvent args)
```
`entity` will cast to EntityUid, so just use that wherever you had uid. You can get the component with Entity.Comp.

**Comment by Brandon-Huu** [comment]
> You should probably share at least some of the ui controls between the criminal records console and the cartridge instead of writing it completely from scratch

Wish I knew how to do that 😭
Maybe it can be done in a refactor once all of the required features are implemented.

**Comment by metalgearsloth** [comment]
Re-open if you come back to it.

**Pull request closed without merging**